### PR TITLE
Specify comment ratio script

### PR DESCRIPTION
**Pull request by TeodorescuIoan**

### CODE REVIEW

**Comment by TeodorescuIoan** [comment]
Here is the output when run on the master branch:

| Comments size | Spec size | Comments-Spec Ratio | Spec-Comments Ratio | Filename |
|:--------------|:----------|:--------------------|:--------------------|:---------|
| 0\t10\t0\t100.00\t./theories/VLSM/Lib/Itauto.v |
| 0\t165\t0\t1650.00\t./theories/VLSM/Core/Equivocation/MsgDepFixedSetEquivocation.v |
| 0\t21\t0\t210.00\t./theories/VLSM/Lib/ListFinSetExtras.v |
| 100\t197\t.50\t1.96\t./theories/VLSM/Core/Equivocation/WitnessedEquivocation.v |
| 102\t255\t.39\t2.49\t./theories/VLSM/Core/ByzantineTraces/FixedSetByzantineTraces.v |
| 111\t240\t.46\t2.16\t./theories/VLSM/Core/TraceableVLSM/MinimalEquivocationTrace.v |
| 113\t115\t.98\t1.01\t./theories/VLSM/Core/ByzantineTraces.v |
| 113\t481\t.23\t4.25\t./theories/VLSM/Core/ELMO/MO.v |
| 132\t417\t.31\t3.15\t./theories/VLSM/Core/Validator.v |
| 13\t250\t.05\t19.08\t./theories/VLSM/Lib/StdppExtras.v |
| 148\t334\t.44\t2.25\t./theories/VLSM/Core/Equivocators/Composition/LimitedEquivocation/FixedEquivocation.v |
| 153\t866\t.17\t5.65\t./theories/VLSM/Core/ELMO/ELMO.v |
| 156\t370\t.42\t2.37\t./theories/VLSM/Core/Equivocation/FixedSetEquivocation.v |
| 160\t551\t.29\t3.44\t./theories/VLSM/Core/MessageDependencies.v |
| 166\t703\t.23\t4.23\t./theories/VLSM/Core/Composition.v |
| 18\t227\t.07\t12.54\t./theories/VLSM/Core/VLSMProjections/VLSMInclusion.v |
| 18\t4\t4.39\t.22\t./theories/VLSM/Lib/EquationsExtras.v |
| 1\t15\t.06\t13.63\t./theories/VLSM/Lib/FinExtras.v |
| 1\t2\t.47\t1.81\t./theories/VLSM/Core/ELMO/AllELMO.v |
| 1\t2\t.47\t1.81\t./theories/VLSM/Lib/SsrExport.v |
| 1\t36\t.02\t32.72\t./theories/VLSM/Lib/Temporal.v |
| 1\t58\t.01\t52.72\t./theories/VLSM/Lib/Measurable.v |
| 1\t60\t.01\t54.54\t./theories/VLSM/Lib/StdppListFinSet.v |
| 1\t78\t.01\t70.90\t./theories/VLSM/Lib/FinSetExtras.v |
| 201\t608\t.33\t3.02\t./theories/VLSM/Core/ELMO/UMO.v |
| 20\t219\t.09\t10.89\t./theories/VLSM/Core/EquivocationProjections.v |
| 211\t1332\t.15\t6.30\t./theories/VLSM/Core/Equivocation.v |
| 22\t98\t.22\t4.43\t./theories/VLSM/Core/Equivocators/Composition/LimitedEquivocation/LimitedEquivocationSimulation.v |
| 24\t116\t.20\t4.81\t./theories/VLSM/Core/VLSMProjections/VLSMPartialProjection.v |
| 26\t142\t.18\t5.44\t./theories/VLSM/Core/Equivocation/TraceWiseEquivocation.v |
| 273\t1224\t.22\t4.48\t./theories/VLSM/Core/VLSM.v |
| 27\t229\t.11\t8.45\t./theories/VLSM/Lib/ListSetExtras.v |
| 27\t60\t.44\t2.21\t./theories/VLSM/Core/ReachableThreshold.v |
| 28\t67\t.41\t2.38\t./theories/VLSM/Core/Equivocation/FullNode.v |
| 2\t18\t.11\t8.57\t./theories/VLSM/Lib/RealsExtras.v |
| 2\t29\t.06\t13.80\t./theories/VLSM/Lib/FinFunExtras.v |
| 2\t83\t.02\t39.52\t./theories/VLSM/Lib/StdppListSet.v |
| 33\t234\t.14\t7.06\t./theories/VLSM/Core/Equivocators/Composition/SimulatingFree/FullReplayTraces.v |
| 33\t330\t.09\t9.96\t./theories/VLSM/Lib/TraceProperties.v |
| 34\t162\t.20\t4.75\t./theories/VLSM/Core/Equivocators/MessageProperties.v |
| 36\t179\t.20\t4.95\t./theories/VLSM/Core/Plans.v |
| 38\t166\t.22\t4.35\t./theories/VLSM/Core/Equivocators/Composition/LimitedEquivocation/LimitedStateEquivocation.v |
| 39\t273\t.14\t6.98\t./theories/VLSM/Lib/Preamble.v |
| 3\t123\t.02\t39.67\t./theories/VLSM/Lib/SortedLists.v |
| 3\t194\t.01\t62.58\t./theories/VLSM/Core/VLSMProjections.v |
| 3\t32\t.09\t10.32\t./theories/VLSM/Lib/TraceClassicalProperties.v |
| 40\t266\t.15\t6.63\t./theories/VLSM/Core/Equivocators/Composition/EquivocatorsComposition.v |
| 41\t186\t.22\t4.52\t./theories/VLSM/Core/Equivocation/LimitedMessageEquivocation.v |
| 43\t126\t.34\t2.92\t./theories/VLSM/Core/Equivocation/NoEquivocation.v |
| 45\t262\t.17\t5.80\t./theories/VLSM/Core/ProjectionTraces.v |
| 49\t130\t.37\t2.64\t./theories/VLSM/Core/Equivocators/Composition/SimulatingFree/SimulatingFree.v |
| 52\t388\t.13\t7.44\t./theories/VLSM/Core/VLSMProjections/VLSMEmbedding.v |
| 55\t570\t.09\t10.34\t./theories/VLSM/Core/VLSMProjections/VLSMTotalProjection.v |
| 55\t98\t.56\t1.77\t./theories/VLSM/Core/Equivocators/Composition/LimitedEquivocation/FixedEquivocationSimulation.v |
| 56\t165\t.33\t2.94\t./theories/VLSM/Core/ByzantineTraces/LimitedByzantineTraces.v |
| 57\t299\t.19\t5.23\t./theories/VLSM/Lib/StreamFilters.v |
| 58\t668\t.08\t11.49\t./theories/VLSM/Core/Equivocators/Composition/EquivocatorsCompositionProjections.v |
| 58\t781\t.07\t13.44\t./theories/VLSM/Lib/ListExtras.v |
| 5\t328\t.01\t64.31\t./theories/VLSM/Lib/StreamExtras.v |
| 5\t80\t.06\t15.68\t./theories/VLSM/Core/Equivocators/EquivocatorReplay.v |
| 63\t415\t.15\t6.57\t./theories/VLSM/Core/VLSMProjections/VLSMStutteringEmbedding.v |
| 68\t284\t.23\t4.17\t./theories/VLSM/Lib/TopSort.v |
| 6\t201\t.02\t32.95\t./theories/VLSM/Core/AnnotatedVLSM.v |
| 71\t287\t.24\t4.03\t./theories/VLSM/Core/ELMO/BaseELMO.v |
| 7\t142\t.04\t20.00\t./theories/VLSM/Lib/NeList.v |
| 7\t318\t.02\t44.78\t./theories/VLSM/Core/Equivocation/MsgDepLimitedEquivocation.v |
| 7\t52\t.13\t7.32\t./theories/VLSM/Lib/NatExtras.v |
| 82\t713\t.11\t8.68\t./theories/VLSM/Core/SubProjectionTraces.v |
| 85\t411\t.20\t4.82\t./theories/VLSM/Core/Equivocators/EquivocatorsProjections.v |
| 89\t563\t.15\t6.31\t./theories/VLSM/Core/Equivocators/Equivocators.v |
| 8\t143\t.05\t17.65\t./theories/VLSM/Core/VLSMProjections/VLSMEquality.v |
| 98\t282\t.34\t2.87\t./theories/VLSM/Core/TraceableVLSM.v |
| 9\t45\t.19\t4.94\t./theories/VLSM/Lib/Traces.v |

**Comment by wkolowski** [comment]
@TeodorescuIoan The output you posted is a malformed. It seems you have `\t` where `|` should be for the table to render correctly.

**Comment by TeodorescuIoan** [comment]
@wkolowski I was just showcasing the output of the script as it is, as it stands it now is not useful for a command line user. If I do that change it is only useful for .md files. Wouldn't it be more useful to make it as a csv file?

**Comment by palmskog** [comment]
@TeodorescuIoan can we add some script option to allow output as Markdown or csv on command? Both are useful in different contexts.

File: TOOLS.md [hunk 1]
```diff
@@ -87,3 +87,20 @@ For example, if you want to limit the number of consecutive newlines to 1, run:
 ```shell
 python scripts/strip-newlines.py 2
 ```
+
+## Analyzing commments and specifications
+
+To analyze comment and specification usage throughout the project (for all `*.v` files), run:
+
+```shell
+./all-comment-ratio.v
```
Comment (palmskog) [inline TOOLS.md:96]: I don't think this is the correct name of the script. The `.v` file extension is for Coq files, and can't be executed standalone. It shouldn't be used for anything but Coq.

File: scripts/all-comment-ratio.sh [hunk 2]
```diff
@@ -1,7 +1,56 @@
 #!/bin/sh
 #| Paper           | Coq                              |
 #|:----------------|:---------------------------------|
```
Comment (wkolowski) [inline scripts/all-comment-ratio.sh:3]: Not sure what this is doing here.

**Review by wkolowski — COMMENTED** [review]
I think you should comment bash scripts rather extensively, as without comments they are mostly unreadable to anyone except their author :)

**Comment by palmskog** [comment]
@TeodorescuIoan nice table, but can you sort it so that highest ratio is on top and then descend to lowest ratio? Easiest if you replace the unsorted comment.

**Pull request merged**